RBAC architecture overview

jwitz opened the pull request:
Resolves https://github.com/astronomer/docs/issues/2321#issuecomment-1630862157

Sadly Mermaid doesn't have the tooling yet for expressing diagrams like this, so it'll have to be an image for now. I know it's not as snazzy as the general architecture, but I think it's better to have less designed diagram than none at all

odaneau-astro commented on static/img/docs/rbac-overview.png:
We could improve the readability & look of this image to have it more in line with the other architecture diagram. I'll share the PS Slide Deck we currently use. Maybe we can copy the style from it.

odaneau-astro reviewed (commented):
The content is good. 
Just one comment on readability.

manmeetkaur commented on astro/astro-architecture.md:
```diff
@@ -20,7 +22,19 @@ Astro Hosted consists of three core components for managing Airflow, all hosted
 
 To learn more about Astro Hybrid architecture and features, see [Astro Hybrid overview](hybrid-overview.md)
 
+## Role based access control (RBAC) architecture
```
Not sure why we have to name this RBAC architecture and not Components of Astro? 
```suggestion
## Components of Astro
```

manmeetkaur commented on astro/astro-architecture.md:
```diff
@@ -20,7 +22,19 @@ Astro Hosted consists of three core components for managing Airflow, all hosted
+
+Astro implements role-based access control for Airflow environments. User roles are scoped to both a Workspace and an Organization:
+
+- A [_Workspace_](manage-workspaces.md) is a collection of Deployments that can be accessed by a specific group of users. Workspaces can be used to group Deployments that share a business use case or environment trait.
+- An _Organization_ is a collection of Workspaces. 
+
+Each Astro user has a Workspace role in each Workspace they belong to, plus a single Organization role. Users can also belong to [Teams](add-user.md#make-a-team), which apply the same role across a group of users. You can create API keys with specific roles to automate most actions on Astro, such as deploying code or managing users. 
+
+Use the following diagram as a reference for how all of these entities interact with each other in Astro.
```
```suggestion
- An Astro _Organization_ is created for you when you subscribes to Astro. It is the top most entity in your Astronomer subscription, where you can invite Users, manage Clusters, monitor usage and more. Each Organization contains Workspaces, Clusters and Users. 

- A [_Workspace_](manage-workspaces.md) is a collection of Deployments that can be accessed by a specific group of users. It is a logical entity that can be used to group Deployments that share a business use case or environment trait. It does not create resources in your Cloud Account. Workspaces can be granted access to Clusters to host Deployments.

- A _Cluster_ is a set of infrastructure resources within your organization’s data plane that hosts Astro Runtime deployments. Clusters are created and managed by Astronomer. An Astro cluster corresponds to a managed  instance of EKS on AWS, AKS on Azure or GKE on GCP. The Astro install typically starts with one cluster for each Astro Organization.

Your organization can choose to configure multiple Astro clusters. This could enable a few benefits, including:
- Clusters in different regions
- Different clusters for development and production environments
- Clusters for specific business needs, such as High Performance Computing or Machine Learning

Most cluster modifications can't be completed in the Cloud UI or with the Astro CLI and require you to contact [Astronomer support](https://support.astronomer.io/).

- An _Astro Deployment_ is an Astro Runtime environment running Apache Airflow. It runs the core components of Apache Airflow, including the Airflow webserver, scheduler, triggerer and one or more workers. After you create a Deployment, you can deploy DAGs to it using Astro CLI or a continuous integration and continuousdelivery (CI/CD) process. All DAGs and tasks on Astro are executed within a Deployment.

Each Deployment has a set of resources attached to it. When creating a new deployment you define the number of schedulers and workers you need, along with their resources. This directly impacts how reliable, scalable, and fast your deployment is. You can have as many deployments as you want as long as you have enough resources in your cluster. For more Information, see [Create a new Deployment on Astro](create-deployment.md)

- Each Astro user has a Workspace role in each Workspace they belong to, plus a single Organization role. Users can also belong to [Teams](add-user.md#make-a-team), which apply the same role across a group of users. You can create API keys with specific roles to automate most actions on Astro, such as deploying code or managing users. 

Use the following diagram as a reference for how all of these components interact with each other in Astro.
```

manmeetkaur reviewed (changes requested):
Thanks for doing this @jwitz ! i was going to work on this and saw your note in time. I made a lot of changes because I think we need a component overview page describing how these components interact and not just a diagram. I took pretty much everything from the PPT that Oliver shared. 

I am also inclined to add the Organizing Workspaces section somewhere in our docs, but as a separate PR.

jwitz commented on astro/astro-architecture.md:
```diff
@@ -20,7 +22,21 @@ Astro Hosted consists of three core components for managing Airflow, all hosted
 
+## Role based access control (RBAC) architecture
+
+Astro implements role-based access control for Airflow environments. User roles are scoped to both a Workspace and an Organization:
```
```suggestion
Astro implements role-based access control for Airflow environments.
```

jwitz commented on astro/astro-architecture.md:
```diff
@@ -20,7 +22,21 @@ Astro Hosted consists of three core components for managing Airflow, all hosted
+All users belong to an _Organization_, which is the highest level in the RBAC system. Each Organization contains a collection of Workspaces, clusters, and users. 
+
+- A [_Workspace_](manage-workspaces.md) is a collection of Deployments that can be accessed by a specific group of users. You can use Workspaces to group Deployments that share a business use case or environment trait.
+- A _cluster_ is a Kubernetes cluster that hosts the infrastructure required to run Deployments. Clusters can host Deployments from multiple Workspaces. 
```
```suggestion
- A _cluster_ is a Kubernetes cluster that hosts the infrastructure required to run Deployments. Clusters can host Deployments from multiple Workspaces. You either use pre-configured clusters on Astro or provision your own dedicated clusters with more specific configurations.
```

jwitz commented:
@manmeetkaur I modified your suggestion in the following ways:

- Removed the Deployment definition (we define it in the general architecture)
- I tried to refocus the content on access control only, because the diagram itself was developed to demonstrate RBAC principles. If you don't think this resolves the original issue of describing other system components (worker node pools, etc), we should address that separately in a more detailed reference architecture.

manmeetkaur reviewed (approved):
LGTM. will create an issue for a separate reference architecture.

jwitz commented on astro/astro-architecture.md:
```diff
@@ -20,7 +22,21 @@ Astro Hosted consists of three core components for managing Airflow, all hosted
 
+## Access control architecture
+
+Astro implements role-based access control for Airflow environments.
```
```suggestion
Astro implements role-based access control (RBAC) for Airflow environments.
```
